presto-native-execution/src/test/java/com/facebook/presto/nativeworker/PrestoNativeQueryRunnerUtils.java

```diff
@@ -92,101 +91,249 @@ public class PrestoNativeQueryRunnerUtils
+    {
+        Map<String, String> extraProperties = new HashMap<>(ImmutableMap.of(
+                "regex-library", "RE2J",
+                "offset-clause-enabled", "true"));
+        Map<String, String> hiveProperties = new HashMap<>(ImmutableMap.of("hive.pushdown-filter-enabled", "true"));
+        return new HiveQueryRunnerBuilder("sql-standard", extraProperties, ImmutableMap.of(), hiveProperties, getNativeWorkerTpcdsProperties(), false);
+    }
+
+    public static class HiveQueryRunnerBuilder
+    {
+        private NativeQueryRunnerParameters nativeQueryRunnerParameters = getNativeQueryRunnerParameters();
+        private Path dataDirectory = nativeQueryRunnerParameters.dataDirectory;
+        private String serverBinary = nativeQueryRunnerParameters.serverBinary.toString();
+        private Integer workerCount = nativeQueryRunnerParameters.workerCount.orElse(4);
+        private Integer cacheMaxSize = 0;
+        private String storageFormat = DEFAULT_STORAGE_FORMAT;
+        private Optional<String> remoteFunctionServerUds = Optional.empty();
+        private Map<String, String> extraProperties;
+        private Map<String, String> extraCoordinatorProperties;
+        private Map<String, String> hiveProperties;
+        private Map<String, String> tpcdsProperties;
+        private String security;
+        private boolean addStorageFormatToPath;
+        private boolean coordinatorSidecarEnabled;
+        private boolean enableRuntimeMetricsCollection;
+        private boolean enableSsdCache;
+        private boolean failOnNestedLoopJoin;
+        // External worker launcher is applicable only for the native query runner and because it depends on other
+        // properties, it should be created once all the other query runner configs are set. This variable indicates
+        // whether the query runner returned by builder should use an external worker launcher, it will be true only for
+        // the native query runner and should NOT be configurable by users.
+        private boolean useExternalWorkerLauncher;
+
+        private HiveQueryRunnerBuilder(String security,
+                                       Map<String, String> extraProperties,
+                                       Map<String, String> extraCoordinatorProperties,
+                                       Map<String, String> hiveProperties,
+                                       Map<String, String> tpcdsProperties,
+                                       boolean useExternalWorkerLauncher)
+        {
+            this.security = security;
+            this.useExternalWorkerLauncher = useExternalWorkerLauncher;
+            this.extraProperties = extraProperties;
+            this.extraCoordinatorProperties = extraCoordinatorProperties;
+            hiveProperties.put("hive.storage-format", storageFormat);
+            this.hiveProperties = hiveProperties;
+            this.tpcdsProperties = tpcdsProperties;
+        }
```

Comment:
Wondering if it's even worth having this constructor. I feel like it makes the other usages of this difficult to read. I think it would be better to have a no-arg constructor and just call `setXXX` appropriately inside the `[java|native]XXXQueryRunner` methods

Reply:
I'm in agreement with this idea in general, though I feel like we probably don't want to have methods like setExtraProperties and setExtraCoordinator properties that are setting the map structures that are closely tied to the structure of the QueryRunnerBuilder.

An option is have a constructor that takes a single input enum with values (JAVA, NATIVE), and use that to initialize the builder with all the properties in nativeHiveQueryRunnerBuilder() or javaHiveQueryRunnerBuilder(). 

We can remove the native(java)HiveQueryRunnerBuilder methods as well and just use this constructor everywhere if possible.

wdyt ?